plugins/main/ipvlan/ipvlan.go

```diff
@@ -246,10 +271,139 @@ func cmdDel(args *skel.CmdArgs) error {
 
 func main() {
 	// TODO: implement plugin version
-	skel.PluginMain(cmdAdd, cmdGet, cmdDel, version.All, "TODO")
+	skel.PluginMain(cmdAdd, cmdCheck, cmdDel, version.All, "TODO")
+}
+
+func cmdCheck(args *skel.CmdArgs) error {
+
+	n, _, err := loadConf(args.StdinData, true)
+	if err != nil {
+		return err
+	}
+	netns, err := ns.GetNS(args.Netns)
+	if err != nil {
+		return fmt.Errorf("failed to open netns %q: %v", args.Netns, err)
+	}
+	defer netns.Close()
+
+	if n.IPAM.Type != "" {
+		// run the IPAM plugin and get back the config to apply
+		err = ipam.ExecCheck(n.IPAM.Type, args.StdinData)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Parse previous result.
+	if n.NetConf.RawPrevResult == nil {
+		return fmt.Errorf("Required prevResult missing")
+	}
+
+	if err := version.ParsePrevResult(&n.NetConf); err != nil {
```

Comment:
I know above there's the shortcut to loadConf() before it does the prevResult processing. I wonder if we could in fact split up loadConf() so that the Master validation happens in a separate function that cmdAdd() would call. While the rest of loadConf() including prevResult handling can be shared with cmdAdd/Del/Check? then we don't have to do the parsing in each of those functoins but can just use n.PrevResult.

Reply:
logic isn't the same for add/check/del